### PR TITLE
chore: update docker template to get and use docker context from Coder host

### PR DESCRIPTION
**Pull request by matifali**
This will allow using the remote docker hosts if there are already set up as the default `docker context` on the Coder host.

Docker provider does support using remote ssh hosts [nativity](https://registry.terraform.io/providers/kreuzwerker/docker/latest/docs#remote-hosts), but that is an extra step.

### Current workflow
1. Login with coder cli from a client machine
2. `coder template init`
3. Choose `docker`
4. Edit the `host` field
5. create the template

### After this change

To use a remote docker host, add and set it using the `docker context` command
1. On the Coder host, run:

```shell
docker context create coder --docker "host=ssh://user@host"
docker context use coder
```
2. Import the starter template from the dashboard; it should work OOB.
Bonus: It will also work if Coder runs on a Windows host with a remote UNIX docker host set up remotely. 

I have tested this on both Linux and Windows (configured to use a Linux remote host) Coder deployments

EDIT: removed `jq` dependence.

### CODE REVIEW

File: examples/templates/docker/main.tf [hunk 1]
```diff
@@ -8,17 +8,36 @@ terraform {
...
+    }
   }
 }
 
 locals {
-  username = data.coder_workspace.me.owner
+  username        = data.coder_workspace.me.owner
+  host_os         = data.coder_provisioner.me.os
+  unix_command    = <<EOF
+  docker context inspect $(docker context show) | grep Host | awk '{print $2}' | tr -d '"' | tr -d ',' | tr -d '\n' | jq -R | jq '{DOCKER_HOST: .}'
```
Comment (mafredri) [inline examples/templates/docker/main.tf:22]: I'm not sure about making our default template dependent on `jq`, seems like a hindrance to get started. 🤔

If we end up using jq anyway, might I suggest a simplification?

```suggestion
  docker context inspect $(docker context show) | jq '{DOCKER_HOST: .[].Endpoints.docker.Host}'
```

Here's an almost "zero dependency" command:

```shell
docker context inspect $(docker context show) | sed -ne 's/.*"Host":\(.*\),$/{"DOCKER_HOST":\1}/p'
```

**Comment by matifali** [comment]
So I removed the `jq` dependency. Thanks to @mafredri for the suggestion. ChatGPT helped for windows case.

**Comment by matifali** [comment]
We can remove this logic when and if the docker provider supports `context` natively.
https://github.com/kreuzwerker/terraform-provider-docker/issues/540

**Comment by matifali** [comment]
It seems like the docker provider will not support the `context` option soon as the docker API doesn't expose it yet. https://docs.docker.com/engine/api/v1.43/

**Comment by matifali** [comment]
> The docs won't work because it needs to be changed as the `coder` user. You can't `su coder` either since we use `/bin/nologin` as the shell. It's possible to do but the commands will be pretty unwieldy

After further investigation, this approach does not work well on Unix. As Coder is running as a separate user can not read the `~/.docker` directory of other users.

**Pull request closed without merging**